components/raftstore-v2/src/operation/query/local.rs

```diff
@@ -215,15 +215,19 @@ where
                         atomic::fence(atomic::Ordering::Release);
                         let snapshot_ts = monotonic_raw_now();
 
-                        if !delegate.is_in_leader_lease(snapshot_ts) {
+                        if !delegate.is_in_leader_lease(snapshot_ts) && !has_read_index_success {
+                            // Redirect if it's not in lease or it has not finish read index.
```

Comment:
is it 'or' or 'and'?

Reply:
Thanks for pointing it out. Will correct comments.

Reply:
```suggestion
                            // Redirect if it's not in lease and it has not finish read index.
```